src/zenml/zen_server/routers/auth_endpoints.py

```diff
@@ -256,6 +259,7 @@ def generate_access_token(
     response_model=Union[OAuthTokenResponse, OAuthRedirectResponse],
 )
 @handle_exceptions
+@limiter.limit("5/minute;1000/day")
```

Comment:
Adding this will throw an exception that is not handled by the `@handle_exceptions` decorator and ends up being incorrectly reported as a 500 internal server error. It should be properly caught and converted into a user-friendly authorization error.

This is what the client sees:

```
RuntimeError: Error initializing rest store with URL 'http://localhost:8237': <slowapi.wrappers.Limit object at 0x7fc80523edc0>
```

This is what the server throws:

```
ratelimit 5 per 1 minute (127.0.0.1) exceeded at endpoint: /api/v1/login
API error
Traceback (most recent call last):
  File "/home/stefan/aspyre/src/zenml/src/zenml/zen_server/utils.py", line 304, in decorated
    return func(*args, **kwargs)
  File "/home/stefan/aspyre/src/zenml/.venv/lib/python3.8/site-packages/slowapi/extension.py", line 764, in sync_wrapper
    self._check_request_limit(request, func, False)
  File "/home/stefan/aspyre/src/zenml/.venv/lib/python3.8/site-packages/slowapi/extension.py", line 630, in _check_request_limit
    self.__evaluate_limits(request, _endpoint_key, all_limits)
  File "/home/stefan/aspyre/src/zenml/.venv/lib/python3.8/site-packages/slowapi/extension.py", line 528, in __evaluate_limits
    raise RateLimitExceeded(failed_limit)
slowapi.errors.RateLimitExceeded: <slowapi.wrappers.Limit object at 0x7fc80523edc0>
INFO:     127.0.0.1:47208 - "POST /api/v1/login HTTP/1.1" 500 Internal Server Error
```

Reply:
Yeah, good catch. I just moved it around and now on UI user will see `5 per 1 minute` and on the server:
```
ratelimit 5 per 1 minute (127.0.0.1) exceeded at endpoint: /api/v1/login
INFO:     127.0.0.1:63151 - "POST /api/v1/login HTTP/1.1" 429 Too Many Requests
```
This seems good enough for me.